Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -449,8 +452,15 @@ void WebFrameProxy::collapseSelection()
 
 void WebFrameProxy::disconnect()
 {
-    if (RefPtr parentFrame = m_parentFrame.get())
+    if (RefPtr parentFrame = m_parentFrame.get()) {
+#if ENABLE(WEBDRIVER_BIDI)
+        if (RefPtr page = m_page.get()) {
+            if (RefPtr session = page->activeAutomationSession())
+                session->contextDestroyedForFrame(*this);
```

Comment:
How is this related to the existing `WebAutomationSession::didDestroyFrame` which does the `m_frameHandleMap` maintenance?